src/REL/mention_detection.py

```diff
@@ -104,28 +129,117 @@ def split_text(self, dataset, is_flair=False):
             splits.append(splits[-1] + i)
         return res, processed_sentences, splits
 
-    def find_mentions(self, dataset, tagger=None):
+
+    def combine_entities(self, ner_results):
+        ner_results_out = []
+        i = 0
+        while i < len(ner_results):
+            last_end = ner_results[i]["end"]
+            ner_results_out.append(dict(ner_results[i]))
+            j = 1
+            while i + j < len(ner_results) and (ner_results[i+j]["start"] == last_end or
+                                                (ner_results[i+j]["start"] == last_end + 1 and 
+                                                 re.search("^I", ner_results[i+j]["entity"]) and
+                                                 re.sub("^..", "", ner_results[i+j]["entity"]) == re.sub("^..", "", ner_results[i]["entity"]))):
+                if ner_results[i+j]["start"] == last_end:
+                    ner_results_out[-1]["word"] += re.sub("^##", "", ner_results[i+j]["word"])
+                else:
+                    ner_results_out[-1]["word"] += " " + ner_results[i+j]["word"]
+                ner_results_out[-1]["end"] = ner_results[i+j]["end"]
+                last_end = ner_results[i+j]["end"]
+                j += 1
+            i += j
+        return ner_results_out
+
+
+    def split_sentence_in_bert_tokens(self, sentence, tagger_ner):
+        tokenizer_results = tagger_ner.tokenizer([sentence], return_offsets_mapping=True) # warns if sentence is too long (>512)
+        input_ids = tokenizer_results["input_ids"][0]
+        token_spans = tokenizer_results["offset_mapping"][0]
+        tokens = [ tagger_ner.tokenizer.decode(token_id) for token_id in input_ids ]
+        return tokens, token_spans
+
+
+
+    def combine_tokens_to_text(self, token_list):
```

Comment:
This method is not used in the code, can it be removed?

Reply:
Correct observation, I have removed the method